packages/block-library/src/quote/theme.scss

```diff
@@ -10,9 +12,21 @@
 		position: relative;
 		font-style: normal;
 	}
-}
 
-.wp-block-quote:not(.is-large):not(.is-style-large) {
-	border-left: 4px solid $black;
-	padding-left: 1em;
+	&[style*="text-align:right"],
+	&[style*="text-align: right"] {
+		border-left: none;
+		border-right: 4px solid $black;
+		padding-left: 0;
+		padding-right: 1em;
+	}
+
+	&[style="text-align:center"] {
```

Comment:
I think this one also needs to include the alternate selector with a space added: `&[style*="text-align: center"]`. I'm not seeing the border disappear otherwise. 

After that's updated, this should be all set.